acceptance/tests/load_libfacter.rb

```diff
@@ -1,31 +1,32 @@
-test_name 'Ruby can load libfacter'
+test_name 'C100161: Ruby can load libfacter' do
+  tag 'risk:high'
 
-agents.each do |agent|
-  on agent, "env PATH=\"#{agent['privatebindir']}:${PATH}\" which ruby"
-  ruby_path = stdout.chomp
+  agents.each do |agent|
+    ruby_path = on(agent, "env PATH=\"#{agent['privatebindir']}:${PATH}\" which ruby").stdout.chomp
 
-  root_dir = ruby_path
-  while File.basename(root_dir).downcase != 'puppet'
-    new_root_dir = File.dirname(root_dir)
-    if new_root_dir == root_dir
-      break
-    else
-      root_dir = new_root_dir
+    root_dir = ruby_path
+    while File.basename(root_dir).downcase != 'puppet'
```

Comment:
The path variable logic here looks brittle to me. If we need to be able to introspect a consistent ruby load path based on installation, then this should probably be added to Beaker.

cc @kevpl , @tvpartytonight